### ISSUE TITLE
Startup errors not propagated properly

### ISSUE THREAD

**Issue by ctron** [eclipse/hono#432]
It looks to me as if startup errors are propagated properly.

The following stack trace shows an exception thrown in "listen", however this error is not reported to the application startup and the application startup times out.

The stack trace shown is created by a manual "catch" which I added:

~~~
13:45:16.830 [vert.x-eventloop-thread-1] INFO  o.e.h.a.m.i.VertxBasedMqttProtocolAdapter - limiting size of inbound message payload to 2048 bytes
13:45:16.833 [vert.x-eventloop-thread-1] INFO  o.e.h.a.m.i.VertxBasedMqttProtocolAdapter - Server uses secure standard port 8883
13:45:16.833 [vert.x-eventloop-thread-1] DEBUG o.e.h.a.m.i.VertxBasedMqttProtocolAdapter - Start binding secure server
13:45:16.841 [vert.x-eventloop-thread-1] DEBUG o.e.h.a.m.i.VertxBasedMqttProtocolAdapter - Dive into listen
13:45:16.877 [vert.x-eventloop-thread-1] ERROR o.e.h.a.m.i.VertxBasedMqttProtocolAdapter - Problem HERE
io.vertx.core.VertxException: io.vertx.core.file.FileSystemException: java.nio.file.NoSuchFileException: /etc/secrets/tls.key
	at io.vertx.core.net.impl.SSLHelper.createContext(SSLHelper.java:305)
	at io.vertx.core.net.impl.SSLHelper.getContext(SSLHelper.java:447)
	at io.vertx.core.net.impl.SSLHelper.validate(SSLHelper.java:467)
	at io.vertx.core.net.impl.NetServerImpl.listen(NetServerImpl.java:177)
	at io.vertx.core.net.impl.NetServerImpl.listen(NetServerImpl.java:320)
	at io.vertx.core.net.impl.NetServerImpl.listen(NetServerImpl.java:336)
	at io.vertx.mqtt.impl.MqttServerImpl.listen(MqttServerImpl.java:95)
	at io.vertx.mqtt.impl.MqttServerImpl.listen(MqttServerImpl.java:69)
	at io.vertx.mqtt.impl.MqttServerImpl.listen(MqttServerImpl.java:74)
	at org.eclipse.hono.adapter.mqtt.AbstractVertxBasedMqttProtocolAdapter.bindMqttServer(AbstractVertxBasedMqttProtocolAdapter.java:175)
	at org.eclipse.hono.adapter.mqtt.AbstractVertxBasedMqttProtocolAdapter.bindSecureMqttServer(AbstractVertxBasedMqttProtocolAdapter.java:133)
	at org.eclipse.hono.adapter.mqtt.AbstractVertxBasedMqttProtocolAdapter.lambda$doStart$3(AbstractVertxBasedMqttProtocolAdapter.java:203)
	at io.vertx.core.Future.lambda$compose$1(Future.java:270)
	at io.vertx.core.impl.FutureImpl.setHandler(FutureImpl.java:81)
	at io.vertx.core.Future.compose(Future.java:266)
	at org.eclipse.hono.adapter.mqtt.AbstractVertxBasedMqttProtocolAdapter.doStart(AbstractVertxBasedMqttProtocolAdapter.java:203)
	at org.eclipse.hono.service.AbstractProtocolAdapterBase.startInternal(AbstractProtocolAdapterBase.java:161)
	at org.eclipse.hono.service.AbstractServiceBase.start(AbstractServiceBase.java:38)
	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$10(DeploymentManager.java:481)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:344)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.vertx.core.file.FileSystemException: java.nio.file.NoSuchFileException: /etc/secrets/tls.key
	at io.vertx.core.file.impl.FileSystemImpl$13.perform(FileSystemImpl.java:745)
	at io.vertx.core.file.impl.FileSystemImpl$13.perform(FileSystemImpl.java:737)
	at io.vertx.core.file.impl.FileSystemImpl.readFileBlocking(FileSystemImpl.java:301)
	at io.vertx.core.net.impl.KeyStoreHelper.create(KeyStoreHelper.java:105)
	at io.vertx.core.net.KeyCertOptions.getKeyManagerFactory(KeyCertOptions.java:47)
	at io.vertx.core.net.impl.SSLHelper.getKeyMgrFactory(SSLHelper.java:310)
	at io.vertx.core.net.impl.SSLHelper.createContext(SSLHelper.java:263)
	... 24 common frames omitted
Caused by: java.nio.file.NoSuchFileException: /etc/secrets/tls.key
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
	at java.nio.file.Files.newByteChannel(Files.java:361)
	at java.nio.file.Files.newByteChannel(Files.java:407)
	at java.nio.file.Files.readAllBytes(Files.java:3152)
	at io.vertx.core.file.impl.FileSystemImpl$13.perform(FileSystemImpl.java:741)
	... 30 common frames omitted
13:45:16.878 [vert.x-eventloop-thread-1] DEBUG o.e.h.a.m.i.VertxBasedMqttProtocolAdapter - Returing to caller
13:45:17.776 [vert.x-eventloop-thread-0] INFO  o.e.h.s.a.d.UsernamePasswordAuthProvider - connected to Credentials service
13:45:36.845 [main] ERROR o.e.h.a.m.i.Application$$EnhancerBySpringCGLIB$$4cb4b561 - startup timed out after 20 seconds, shutting down ...
13:45:36.849 [main] DEBUG o.e.h.a.m.i.Application$$EnhancerBySpringCGLIB$$4cb4b561 - shutting down application...
13:45:36.898 [vert.x-eventloop-thread-0] DEBUG o.e.hono.client.impl.HonoClientImpl - lost connection to server [hono-service-device-registry.hono.svc:5671]
13:45:36.908 [main] INFO  o.e.h.a.m.i.Application$$EnhancerBySpringCGLIB$$4cb4b561 - application has been shut down successfully
~~~

**Comment by ctron** [comment]
So after a lot of debugging I do think this is a regression in vertx 3.5.0. It seems to work in vertx 3.4.1. I will write this up and create an issue with vertx.

**Comment by ctron** [comment]
Reported as eclipse/vert.x#2234

**Comment by ctron** [comment]
It looks like this will be fixed in Vertx 3.5.1.

**Comment by sophokles73** [comment]
Thanks, @ctron :+1:

**Comment by ctron** [comment]
No, because the problem still exists. We can't do much except downgrading vertx. So I would suggest keeping this open until we verified eclipse/vert.x#2234 is fixed and we upgraded to vertx 3.5.1.

**Comment by sophokles73** [comment]
I see.

**Comment by sophokles73** [comment]
created #529 for upgrading to vert.x 3.5.1

**Comment by ctron** [comment]
I guess this can be closed now as vertx is upgraded by commit 234940950a41d2744509e83e4823e9e0fd0e9f45

**Comment by sophokles73** [comment]
@ctron  Then be my guest to do so :-)
And please also update the issue's milestone and project to 0.6-M2.
